rfd/xxxx-config-resource-docs.md

```diff
@@ -0,0 +1,554 @@
+   decl  *ast.Decl // Declaration data
+   file  string   // File path where the declaration was made
+  }
+  ```
+
+- Final data map: `map[declKey]Resource`: Tracking type data to use to populate
+  a template. This ensures that each declaration is only included once.
+
+In each `*packages.Package` in the slice mentioned earlier, the generator ranges
+through the package's `Syntax` field, which is an `[]*ast.File`. Within each
+`*ast.File`, the generator ranges through the file's
+[`Decls`](https://pkg.go.dev/go/ast#File) field and adds each declaration to the
+declaration map. The `CompiledGoFiles` field in `packages.Package` lists paths
+of Go source files in the same order as the `Syntax` field, so we can use this
+to populate the `file` property of each `sourceDecl` in the map.
+
+The next step is to look up each user-configured struct type within the
+declaration map and begin generating a `Resource` for that struct type, plus
+additional `Resource` structs for the types of its fields (and their types
+and so on). The sequence is:
+
+- Look up a user-defined struct type from the configuration within the
+  declaration map.
+- Construct a `Resource` from the struct type and insert it into the final data
+  map (unless a `Resource` already exists there).
+- Process the fields of the user-defined struct type using the rules described
+  [below](#processing-field-types).
+
+Once this is done, the generator will convert the  `map[declKey]Resource` into a
+single `[]Resource` to feed into the template.
+
+#### Alternative: Using analysis drivers
+
+The Go Tools repository provides the `multichecker`
+(https://pkg.go.dev/golang.org/x/tools/go/analysis/multichecker) and
+`singlechecker`
+(https://pkg.go.dev/golang.org/x/tools/go/analysis/singlechecker) packages
+within `golang.org/x/tools/go/analysis`, which run custom static analysis
+functions, `analysis.Analyzer`s. However, these static analysis tools are
+intended for linting, not generating reference documentation. 
+
+While you can configure an `analysis.Analyzer` to return a value from its `Run`
+function, the `multichecker` and `singlechecker` drivers run by exposing a
+`Main` function that handles all of the program's output and CLI arguments. It
+is not possible to initialize a data structure outside the `Main` function and
+assign values to it during the `Run` function, since `Main`
+[exits](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.8.0:go/analysis/multichecker/multichecker.go;l=59)
+the current process after completing, leaving the program no chance to process
+the data structure.
+
+The `singlechecker` and `multichecker` packages run static analysis by importing
+from a `checker` package, which is internal to the
+`analysis` package. While we could copy this code and
+modify it for our purposes, the reference generator does not require the
+general-purpose static analysis capabilities of the `analysis` package.
+
+### Converting structs to `Resource`s
```

Comment:
For now, I think the only requirement is that a GoDoc exists. I've suggested exiting with an error if there's no GoDoc above a field so we can add one. 

The only special rule is that, if a field name begins with the name of the field, remove the field name (and "is" if it exists) and capitalize the next first letter. This is to prevent field names that aren't user facing from making it into the docs, and also removes repetition with the first column of the table.

This GoDoc:

```go
// MyField is a field that specifies something.
```

...would become:

```
A field that specifies something
```

Otherwise, I think it's okay to move each field's GoDoc into the table without alteration.

Reply:
> if a field name begins with the name of the field, remove the field name

The caveat I see here is that sometimes the field name is part of the sentence in a way that could have this break, but I suppose we can be careful and edit the types we care about.

Reply:
> The caveat I see here is that sometimes the field name is part of the sentence in a way that could have this break, but I suppose we can be careful and edit the types we care about.

We could also simply replace all mentions of the original field name in the GoDoc for a field with mentions of the one included in the docs, i.e., the struct tag. I'll include that in the RFD.